### PR TITLE
Add has view support to blueprinter

### PR DESCRIPTION
**Pull request by spencerneste**

### CODE REVIEW

File: lib/blueprinter/base.rb [hunk 1]
```diff
@@ -431,5 +431,26 @@ def self.view(view_name)
...
+    #
+    # @example With the following Blueprinter
+    #
+    # class ExampleBlueprinter < Blueprinter::Base
+    #  view :custom do
+    #  end
+    # end
+    #
+    #  ExampleBlueprinter.has_view?(:custom) => true
+    #  ExampleBlueprinter.has_view?(:doesnt_exist) => false
```
Comment (mcclayton) [inline lib/blueprinter/base.rb:448]:
```suggestion
    # class ExampleBlueprint < Blueprinter::Base
    #  view :custom do
    #  end
    # end
    #
    #  ExampleBlueprint.has_view?(:custom) => true
    #  ExampleBlueprint.has_view?(:doesnt_exist) => false
```

File: lib/blueprinter/base.rb [hunk 2]
```diff
@@ -431,5 +431,26 @@ def self.view(view_name)
       yield
       @current_view = view_collection[:default]
     end
+
+    # Check whether or not a Blueprint supports the supplied view.
+    # It accepts a view name.
+    #
+    # @param view_name [Symbol] the view name
+    #
+    # @example With the following Blueprinter
```
Comment (mcclayton) [inline lib/blueprinter/base.rb:440]:
```suggestion
    # @example With the following Blueprint
```

File: lib/blueprinter/base.rb [hunk 3]
```diff
@@ -431,5 +431,26 @@ def self.view(view_name)
...
+    #
+    # class ExampleBlueprinter < Blueprinter::Base
```
Comment (mcclayton) [inline lib/blueprinter/base.rb:442]:
```suggestion
    # class ExampleBlueprint < Blueprinter::Base
```

File: lib/blueprinter/base.rb [hunk 4]
```diff
@@ -431,5 +431,26 @@ def self.view(view_name)
...
+    #  view :custom do
+    #  end
+    # end
+    #
+    #  ExampleBlueprinter.has_view?(:custom) => true
+    #  ExampleBlueprinter.has_view?(:doesnt_exist) => false
+    #
+    # @return [Boolean] a boolean value indicating if the view is
+    # suppored by this Blueprinter.
```
Comment (mcclayton) [inline lib/blueprinter/base.rb:451]:
```suggestion
    # suppored by this Blueprint.
```

File: spec/integrations/base_spec.rb [hunk 5]
```diff
@@ -353,6 +353,28 @@ def extract(association_name, object, _local_options, _options={})
...
+        end
+      end
+    end
+
+    context 'when the blueprinter has the supplied view' do
+      let(:view) { :custom }
+      it { is_expected.to eq(true) }
+    end
+
+    context 'when the blueprinter does not have the supplied view' do
```
Comment (mcclayton) [inline spec/integrations/base_spec.rb:372]:
```suggestion
    context 'when the blueprint has the supplied view' do
      let(:view) { :custom }
      it { is_expected.to eq(true) }
    end

    context 'when the blueprint does not have the supplied view' do
```

File: lib/blueprinter/base.rb [hunk 6]
```diff
@@ -431,5 +431,26 @@ def self.view(view_name)
...
+    # class ExampleBlueprint < Blueprinter::Base
+    #  view :custom do
+    #  end
+    # end
+    #
+    #  ExampleBlueprint.has_view?(:custom) => true
+    #  ExampleBlueprint.has_view?(:doesnt_exist) => false
+    #
+    # @return [Boolean] a boolean value indicating if the view is
+    # suppored by this Blueprint.
```
Comment (cagmz) [inline lib/blueprinter/base.rb:451]:
```suggestion
    # supported by this Blueprint.
```

**Comment by spencerneste** [comment]
@mcclayton Thanks for the review! Any requirements for testing? Specs are green , but is there anything else needed?

**Comment by mcclayton** [comment]
I think it's all good to merge, but might be nice for @philipqnguyen to take a look to make sure this is something we want to expose, but I think it's good 👍

**Pull request merged**